code/datums/elements/light_eater.dm

```diff
@@ -0,0 +1,187 @@
+ *
+ * Arguments:
+ * - [source][/obj/item]: The item what was used to strike the target
+ * - [target][/atom]: The atom being struck by the user with the source
+ * - [user][/mob/living]: The mob using the source to strike the target
+ * - proximity: Whether the strike was in melee range so you can't eat lights from cameras
+ */
+/datum/element/light_eater/proc/on_afterattack(obj/item/source, atom/target, mob/living/user, proximity)
+	SIGNAL_HANDLER
+	if(!proximity)
+		return NONE
+	eat_lights(target, source)
+	return COMPONENT_AFTERATTACK_PROCESSED_ITEM
+
+/**
+ * Called when a source object is used to block a thrown object, projectile, or attack
+ *
+ * Arguments:
+ * - [source][/obj/item]: The item what was used to block the target
+ * - [owner][/mob/living/carbon/human]: The mob that blocked the target with the source
+ * - [hitby][/atom/movable]: The movable that was blocked by the owner with the source
+ * - attack_text: The text tring that will be used to report that the target was blocked
+ * - final_block_chance: The probability of blocking the target with the source
+ * - attack_type: The type of attack that was blocked
+ */
+/datum/element/light_eater/proc/on_hit_reaction(obj/item/source, mob/living/carbon/human/owner, atom/movable/hitby, attack_text, final_block_chance, damage, attack_type, damage_type)
+	SIGNAL_HANDLER
+	if(prob(final_block_chance))
+		eat_lights(hitby, source)
+	return NONE
+
+/**
+ * Called when a produced projectile strikes a target atom
+ *
+ * Arguments:
+ * - [source][/datum]: The thing that created the projectile
+ * - [firer][/atom/movable]: The movable atom that fired the projectile
+ * - [target][/atom]: The atom that was struck by the projectile
+ * - angle: The angle the target was struck at
+ */
+/datum/element/light_eater/proc/on_projectile_hit(datum/source, atom/movable/firer, atom/target, angle)
+	SIGNAL_HANDLER
+	eat_lights(target, source)
+	return NONE
+
+/**
+ * Called when a source projectile strikes a target atom
+ *
+ * Arguments:
+ * - [source][/obj/projectile]: The projectile striking the target atom
+ * - [firer][/atom/movable]: The movable atom that fired the projectile
+ * - [target][/atom]: The atom that was struck by the projectile
+ * - angle: The angle the target was struck at
+ * - hit_limb: The limb that was hit, if the target was a carbon
+ */
+/datum/element/light_eater/proc/on_projectile_self_hit(obj/projectile/source, atom/movable/firer, atom/target, angle, hit_limb)
+	SIGNAL_HANDLER
+	eat_lights(target, source)
+	return NONE
+
+/**
+ * Called when a source reagent exposes a target atom
+ *
+ * Arguments:
+ * - [source][/datum/reagent]: The reagents that exposed the target atom
+ * - [target][/atom]: The atom that was exposed to the light reater reagents
+ * - reac_volume: The volume of the reagents the target was exposed to
+ * 
+ * 
+/datum/element/light_eater/proc/on_expose_atom(datum/reagent/source, atom/target, reac_volume)
```

Comment:
it was a proc that tg has, but we don't yet, rather than excluding it i thought i'd just comment it out
basically, if a chemical was added and given light eater, you'd be able to splash it on someone for it take effect